x-pack/test/defend_workflows_cypress/fleet_server.ts

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { ChildProcess, spawn } from 'child_process';
+import { ToolingLog } from '@kbn/tooling-log';
+import axios, { AxiosRequestConfig } from 'axios';
+import { Manager } from './resource_manager';
+import { getLatestVersion } from './artifact_manager';
+
+export interface AgentManagerParams {
+  user: string;
+  password: string;
+  kibanaUrl: string;
+  esHost: string;
+  esPort: string;
+}
+
+export class FleetManager extends Manager {
+  private fleetProcess?: ChildProcess;
+  private config: AgentManagerParams;
+  private log: ToolingLog;
+  private requestOptions: AxiosRequestConfig;
+  constructor(config: AgentManagerParams, log: ToolingLog, requestOptions: AxiosRequestConfig) {
+    super();
+    this.config = config;
+    this.log = log;
+    this.requestOptions = requestOptions;
+  }
+  public async setup(): Promise<void> {
+    this.log.info('Setting fleet up');
+    return new Promise(async (res, rej) => {
+      try {
+        // default fleet server policy no longer created by default
+        const {
+          data: {
+            item: { id: policyId },
+          },
+        } = await axios.post(
+          `${this.config.kibanaUrl}/api/fleet/agent_policies`,
+          {
+            name: 'Default Fleet Server policy',
+            description: '',
+            namespace: 'default',
+            monitoring_enabled: ['logs', 'metrics'],
+            has_fleet_server: true,
+          },
+          this.requestOptions
+        );
+
+        const response = await axios.post(
+          `${this.config.kibanaUrl}/api/fleet/service_tokens`,
+          {},
+          this.requestOptions
+        );
+        const serviceToken = response.data.value;
+        const artifact = `docker.elastic.co/beats/elastic-agent:${await getLatestVersion()}`;
+        this.log.info(artifact);
+
+        const host = 'host.docker.internal';
+
+        const args = [
+          'run',
+          '-p',
+          `8220:8220`,
```

Comment:
we probably want to limit this to localhost, docker will overwrite host firewall rules otherwise - ref https://docs.docker.com/engine/reference/commandline/run/#publish

```suggestion
          `127.0.0.1:8220:8220`,
```

Reply:
Thank you @jbudz !